src/core_test/mpas_test_core_streams.F

```diff
@@ -391,6 +391,197 @@ subroutine test_core_streams_test(domain, threadErrs, ierr)
+   !>
+   !> \details This subroutine calls the `test_remove_alarm` subroutine,
+   !>          passing the `MPAS_STREAM_OUTPUT` as the stream type to test
+   !>          the functionality of removing an output alarm from a stream.
+   !
+   subroutine test_remove_output_alarm(domain, threadErrs, ierr)
+      use mpas_stream_manager
+      implicit none
+      type(domain_type), intent(inout) :: domain
+      integer, dimension(:), intent(out) :: threadErrs
+      integer, intent(out) :: ierr
+
+      threadErrs = 0
+      call test_remove_alarm(domain, threadErrs, ierr, MPAS_STREAM_OUTPUT)
+   end subroutine test_remove_output_alarm
+
+
+   subroutine test_remove_input_alarm(domain, threadErrs, ierr)
+      use mpas_stream_manager
+      implicit none
+      type(domain_type), intent(inout) :: domain
+      integer, dimension(:), intent(out) :: threadErrs
+      integer, intent(out) :: ierr
+
+      threadErrs = 0
+      call test_remove_alarm(domain, threadErrs, ierr, MPAS_STREAM_INPUT)
+   end subroutine test_remove_input_alarm
+
+   !***********************************************************************
+   !  Subroutine test_remove_input_alarm
+   !
+   !> \brief   Wrapper for testing the removal of an input alarm from a stream.
+   !>
+   !> \details This subroutine calls the `test_remove_alarm` subroutine,
+   !>          passing the `MPAS_STREAM_INPUT` as the stream type to test
+   !>          the functionality of removing an input alarm from a stream.
+   !
+   subroutine test_core_streams_test(domain, threadErrs, ierr)
+
+      use mpas_log
+
+      implicit none
+
+      type (domain_type), intent(inout) :: domain
+      integer, dimension(:), intent(out) :: threadErrs
+      integer, intent(out) :: ierr
+      integer :: ierr1, ierr2, ierr3
+
+      ierr = 0
+      ierr1 = 0
+      ierr2 = 0
+      ierr3 = 0
+
+      call mpas_log_write('Testing reading/writing real-valued streams')
+
+      call test_read_write_real_streams(domain, threadErrs, ierr1)
+
+      if (ierr1 == 0) then
+         call mpas_log_write('Stream I/O tests: SUCCESS')
+      else
+         call mpas_log_write('Stream I/O tests: FAILURE', MPAS_LOG_ERR)
+      end if
+
+      call mpas_log_write('Testing removing an output alarm from a stream.')
+      call test_remove_output_alarm(domain, threadErrs, ierr2)
+      if (ierr2 == 0) then
+         call mpas_log_write('Removing output alarm test: SUCCESS')
+      else
+         call mpas_log_write('Removing output alarm test: FAILURE', MPAS_LOG_ERR)
+      end if
+
+      call mpas_log_write('Testing removing an input alarm from a stream.')
+      call test_remove_input_alarm(domain, threadErrs, ierr)
+      if (ierr3 == 0) then
+         call mpas_log_write('Removing input alarm test: SUCCESS')
+      else
+         call mpas_log_write('Removing input alarm test: FAILURE', MPAS_LOG_ERR)
+      end if
+
+      ierr = abs(ierr1) + abs(ierr2) + abs(ierr3)
```

Comment:
Rather than having a new `ierrX` variable for every test, I think we could just use a local `ierr` variable and immediately add its absolute value to the return `ierr` after each test call.